src/range_proof.rs

```diff
@@ -890,6 +895,16 @@ where
             extension_degree,
         })
     }
+
+    /// Helper function to return the serialized proof's extension degree
+    pub fn extension_degree_from_proof_bytes(slice: &[u8]) -> Result<ExtensionDegree, ProofError> {
+        if slice.is_empty() || (slice.len() - 1) % 32 != 0 {
+            return Err(ProofError::InvalidLength(
+                "Invalid serialized proof bytes length".to_string(),
+            ));
+        }
+        ExtensionDegree::try_from(read8(&slice[0..])[0] as usize)
```

Comment:
This surely only needs to be a byte

Reply:
Yip, actually the name is confusing. Fixed the name.